pkgs/development/python-modules/devito/default.nix

```diff
@@ -0,0 +1,68 @@
+{ stdenv
+, lib
+, buildPythonPackage
+, fetchFromGitHub
+, anytree
+, pytest-cov
+, codecov
+, nbval
+, flake8
+, numpy
+, sympy
+, scipy
+, cached-property
+, psutil
+, py-cpuinfo
+, cgen
+, click
+, multidict
+, distributed
+, pytest-runner
+, pyrevolve
+, codepy
+}:
+
+buildPythonPackage rec {
+  pname = "devito";
+  # grabbing the commit that has NixOS compatibility fix
+  version = "4.6.2-bbad3b1";
+
+  src = fetchFromGitHub {
+    repo = "devito";
+    owner = "devitocodes";
+    rev = "bbad3b17be8c9061ca1967760827b2e4b1f11afe";
+    sha256 = "1yd89jh897fkc7089smgnxmgkvwkd0bsap1i93ixmh7pwzjsbdm3";
+  };
+
+  # several tests fail but they pass outside the build (I've tested it); they are also very slow.
+  doCheck = false;
+
+  propagatedBuildInputs = [
```

Comment:
Please clear out this list! There are duplicates and things like pytest-cov and codecov which are normally would go into checkInputs but we remove them all the time because they slow down testing.

It does not matter if upstream defines the dependencies in their requirements.txt if they are obviously unused or wrong.

Reply:
How do we remove them? Pacthing the setup.py?

Reply:
yes, or setup.cfg or requirements.txt. Depending how the project works.